### PR TITLE
[expo-updates] Fix asset hash validation

### PR DESCRIPTION
**Pull request by wschurman**
# Why

base64url encoding is case sensitive, so it doesn't make sense to lowercase it when verifying asset hashes.

This will need to be cherry-picked into the sdk-45 branch.

# How

On android, this was broken since only one side of the comparison was being lower cased. The fix was to remove all lower-casing.
On iOS, this was broken for two reasons:
- We weren't verifying it in all the places we download assets
- We were lower casing both sides of the comparison, which although works is technically not correct since there would now be two hashes that are equivalent (though impossible to find).

# Test Plan

1. Breakpoint on both iOS and android in the asset downloading code.
2. Load both an experience without and with expected asset hashes (legacy manifest and new manifest respectively)
3. Ensure validation is run and works.

### CODE REVIEW

File: packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesRemoteAppLoader.m [hunk 1]
```diff
@@ -104,7 +105,15 @@ - (void)downloadAsset:(EXUpdatesAsset *)asset
                                 extraHeaders:asset.extraRequestHeaders ?: @{}
                                 successBlock:^(NSData *data, NSURLResponse *response) {
         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-          [self handleAssetDownloadWithData:data response:response asset:asset];
+          NSString *hashBase64String = [EXUpdatesUtils base64UrlEncodedSHA256WithData:data];
+          if (asset.expectedHash && ![asset.expectedHash isEqualToString:hashBase64String]) {
+            NSError* error = [NSError errorWithDomain:EXUpdatesRemoteAppLoaderErrorDomain
```
Comment (ide) [inline packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesRemoteAppLoader.m:110]:
```suggestion
            NSError *error = [NSError errorWithDomain:EXUpdatesRemoteAppLoaderErrorDomain
```

**Review by ide — APPROVED** [review]

**Review by esamelson — APPROVED** [review]
Confirmed the Android fix works on my end as well.

File: ios/versioned/sdk43/EXUpdates/EXUpdates/AppLoader/ABI43_0_0EXUpdatesRemoteAppLoader.m [hunk 2]
```diff
@@ -104,7 +105,15 @@ - (void)downloadAsset:(ABI43_0_0EXUpdatesAsset *)asset
                                 extraHeaders:asset.extraRequestHeaders ?: @{}
                                 successBlock:^(NSData *data, NSURLResponse *response) {
         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-          [self handleAssetDownloadWithData:data response:response asset:asset];
+          NSString *hashBase64String = [ABI43_0_0EXUpdatesUtils base64UrlEncodedSHA256WithData:data];
+          if (asset.expectedHash && ![asset.expectedHash isEqualToString:hashBase64String]) {
+            NSError* error = [NSError errorWithDomain:ABI43_0_0EXUpdatesRemoteAppLoaderErrorDomain
```
Comment (wschurman) [inline ios/versioned/sdk43/EXUpdates/EXUpdates/AppLoader/ABI43_0_0EXUpdatesRemoteAppLoader.m:110]:
```suggestion
            NSError *error = [NSError errorWithDomain:ABI43_0_0EXUpdatesRemoteAppLoaderErrorDomain
```

File: ios/versioned/sdk44/EXUpdates/EXUpdates/AppLoader/ABI44_0_0EXUpdatesRemoteAppLoader.m [hunk 3]
```diff
@@ -104,7 +105,15 @@ - (void)downloadAsset:(ABI44_0_0EXUpdatesAsset *)asset
                                 extraHeaders:asset.extraRequestHeaders ?: @{}
                                 successBlock:^(NSData *data, NSURLResponse *response) {
         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-          [self handleAssetDownloadWithData:data response:response asset:asset];
+          NSString *hashBase64String = [ABI44_0_0EXUpdatesUtils base64UrlEncodedSHA256WithData:data];
+          if (asset.expectedHash && ![asset.expectedHash isEqualToString:hashBase64String]) {
+            NSError* error = [NSError errorWithDomain:ABI44_0_0EXUpdatesRemoteAppLoaderErrorDomain
```
Comment (wschurman) [inline ios/versioned/sdk44/EXUpdates/EXUpdates/AppLoader/ABI44_0_0EXUpdatesRemoteAppLoader.m:110]:
```suggestion
            NSError *error = [NSError errorWithDomain:ABI44_0_0EXUpdatesRemoteAppLoaderErrorDomain
```

File: ios/versioned/sdk45/EXUpdates/EXUpdates/AppLoader/ABI45_0_0EXUpdatesRemoteAppLoader.m [hunk 4]
```diff
@@ -104,7 +105,15 @@ - (void)downloadAsset:(ABI45_0_0EXUpdatesAsset *)asset
                                 extraHeaders:asset.extraRequestHeaders ?: @{}
                                 successBlock:^(NSData *data, NSURLResponse *response) {
         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-          [self handleAssetDownloadWithData:data response:response asset:asset];
+          NSString *hashBase64String = [ABI45_0_0EXUpdatesUtils base64UrlEncodedSHA256WithData:data];
+          if (asset.expectedHash && ![asset.expectedHash isEqualToString:hashBase64String]) {
+            NSError* error = [NSError errorWithDomain:ABI45_0_0EXUpdatesRemoteAppLoaderErrorDomain
```
Comment (wschurman) [inline ios/versioned/sdk45/EXUpdates/EXUpdates/AppLoader/ABI45_0_0EXUpdatesRemoteAppLoader.m:110]:
```suggestion
            NSError *error = [NSError errorWithDomain:ABI45_0_0EXUpdatesRemoteAppLoaderErrorDomain
```

**Pull request merged**